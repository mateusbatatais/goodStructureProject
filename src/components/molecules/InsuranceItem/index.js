import React, { useState } from "react";
import handleDate from "../../../utils/handleDateString";
import handleRenewDate from "../../../utils/handleRenewDate";
import styles from "./styles.module.scss";

const InsuranceItem = ({ item }) => {
  const [active, setActive] = useState(false);

  return (
    <article
      data-testid="item-click"
      onClick={() => setActive(!active)}
      className={`${styles.card} ${
        active ? styles.active : ""
      } rounded d-flex my-4 p-4`}
    >
      <div className="flex-fill">
        <section
          className={`${styles.cardTitleHolder} border-bottom d-flex justify-content-between`}
        >
          <i
            data-testid="arrow"
            className={`d-none d-md-block rounded-circle me-4 flex-shrink-0 flex-grow-0 ${
              styles.cardIcon
            } ${active ? styles.cardIconActive : ""}`}
          ></i>
          <div className="flex-fill">
            <h3 className={`${styles.cardTitle} mb-0`}>{item.title}</h3>
            <div className={`${styles.cardSubTitle}`}>
              {item.id} | {item.description}
            </div>
          </div>
          <img
            className={`${styles.cardImageTablet} d-none d-md-block d-lg-none`}
            src={item.partner && item.partner.logo}
            alt={item.partner && item.partner.name}
            title={item.partner && item.partner.name}
          />
        </section>
        <section className={`${styles.cardDetails} d-flex`}>
          <div className="d-none d-md-block">
            <div className={styles.cardTitleDetail}>
              {handleDate(item.payment_date)}
            </div>
            <div className={styles.cardSubTitleDetail}>Payment date</div>
          </div>
          <div>
            <div className={styles.cardTitleDetail}>
              {handleDate(item.coverage_start_date)}
              {item.status === "active" && (
                <> to {handleDate(item.coverage_end_date)}</>
              )}
            </div>
            <div className={styles.cardSubTitleDetail}>
              {item.status === "active" ? "Coverage dates" : "Date shipped"}
            </div>
          </div>
          <div className="d-none d-md-block">
            <div className={styles.cardTitleDetail}>
              {item.premium_formatted}
            </div>
            <div className={styles.cardSubTitleDetail}>Price/Premium</div>
          </div>
          {item.renewal && (
            <div className="d-none d-md-block">
              <div className={styles.cardTitleDetail}>
                {handleRenewDate(item.renewal, item.payment_date)}
              </div>
              <div className={styles.cardSubTitleDetail}>Renewal date</div>
            </div>
          )}
          <div className="d-md-none text-end flex-fill">
            <img
              className={styles.cardImageMob}
              src={item.partner && item.partner.logo}
              alt={item.partner && item.partner.name}
              title={item.partner && item.partner.name}
            />
          </div>
        </section>
      </div>

      <img
        className={`${styles.cardImageDesktop} d-none d-lg-block ms-4`}
        src={item.partner && item.partner.logo}
        alt={item.partner && item.partner.name}
        title={item.partner && item.partner.name}
      />
    </article>
  );
};

export default InsuranceItem;
