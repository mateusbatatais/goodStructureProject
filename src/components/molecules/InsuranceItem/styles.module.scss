@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";
@import "~bootstrap/scss/mixins/_breakpoints";

@import "../../../css/mixins";
@import "../../../css/vars";

.card {
  background-color: white;
  @include transition;
}
.cardIcon {
  border: 2px solid $yellow;
  height: 48px;
  width: 48px;
  background: no-repeat center center
    url("../../../assets/icons/chevron-right-solid.svg");
  background-size: 20%;
  @include media-breakpoint-up(lg) {
    background-size: 30%;
  }
  @include transition;
}
.active {
  box-shadow: 0px 24px 32px rgba(45, 45, 45, 0.16);
}
.cardIconActive {
  background-color: $yellow;
  border: 2px solid $bege;
  @include media-breakpoint-up(lg) {
    background-size: 20%;
  }
  transform: rotate(90deg);
}
.cardTitleHolder {
  margin-bottom: 8px;
  padding-bottom: 8px;
  @include media-breakpoint-up(md) {
    margin-bottom: 16px;
    padding-bottom: 16px;
  }
}
.cardTitle {
  font-weight: 700;
  font-size: 18px;
  line-height: 22.5px;
  @include media-breakpoint-up(md) {
    font-size: 20px;
    line-height: 30px;
  }
  @include media-breakpoint-up(lg) {
    font-size: 24px;
  }
}
.cardSubTitle {
  font-family: "Bitter";
  font-style: normal;
  font-weight: 400;
  line-height: 21px;
  color: $text-clear;
  font-size: 14px;
}
.cardTitleDetail {
  font-weight: 390;
  font-size: 16px;
  line-height: 24px;
}
.cardSubTitleDetail {
  font-family: "Bitter";
  color: $text-clear;
}
.cardDetails {
  > div:not(:first-child) {
    @include media-breakpoint-up(md) {
      border-left: 1px solid #dee2e6;
      margin-left: 1.5rem;
      padding-left: 1.5rem;
    }
  }
}
.cardImageDesktop {
  width: 176px;
}
.cardImageTablet {
  width: 95px;
}
.cardImageMob {
  width: 80px;
}
