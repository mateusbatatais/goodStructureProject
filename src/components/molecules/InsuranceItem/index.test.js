import { render, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import InsuranceItem from ".";
import data from "./mocks/data";

describe("Show Insurances item", () => {
  test("snapshot", () => {
    const container = render(<InsuranceItem item={data} />);
    expect(container).toMatchSnapshot();
  });

  test("Check if click add shadow", async () => {
    const { getAllByTestId } = render(<InsuranceItem item={data} />);
    const itemClicked = getAllByTestId("item-click");
    await userEvent.click(itemClicked[0]);
    await waitFor(() => {
      expect(itemClicked[0].classList.contains("active")).toBe(true);
    });
  });

  test("Check if click change arrow", async () => {
    const { getAllByTestId } = render(<InsuranceItem item={data} />);
    const itemClicked = getAllByTestId("item-click");
    const arrow = getAllByTestId("arrow");
    await userEvent.click(itemClicked[0]);
    await waitFor(() => {
      expect(arrow[0].classList.contains("cardIconActive")).toBe(true);
    });
  });
});
