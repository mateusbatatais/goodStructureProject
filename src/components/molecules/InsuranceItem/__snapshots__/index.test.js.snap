// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Insurances item snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="card  rounded d-flex my-4 p-4"
        data-testid="item-click"
      >
        <div
          class="flex-fill"
        >
          <section
            class="cardTitleHolder border-bottom d-flex justify-content-between"
          >
            <i
              class="d-none d-md-block rounded-circle me-4 flex-shrink-0 flex-grow-0 cardIcon "
              data-testid="arrow"
            />
            <div
              class="flex-fill"
            >
              <h3
                class="cardTitle mb-0"
              >
                London to Paris
              </h3>
              <div
                class="cardSubTitle"
              >
                TRAVEL-COVER-INS
                 | 
                Baggage Cover; Medical Cover; Cancellation Cover
              </div>
            </div>
            <img
              alt="Labore Inc."
              class="cardImageTablet d-none d-md-block d-lg-none"
              src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
              title="Labore Inc."
            />
          </section>
          <section
            class="cardDetails d-flex"
          >
            <div
              class="d-none d-md-block"
            >
              <div
                class="cardTitleDetail"
              >
                10-MAY-2019
              </div>
              <div
                class="cardSubTitleDetail"
              >
                Payment date
              </div>
            </div>
            <div>
              <div
                class="cardTitleDetail"
              >
                16-JUL-2019
                 to 
                18-FEB-2019
              </div>
              <div
                class="cardSubTitleDetail"
              >
                Coverage dates
              </div>
            </div>
            <div
              class="d-none d-md-block"
            >
              <div
                class="cardTitleDetail"
              >
                AUD $106.65
              </div>
              <div
                class="cardSubTitleDetail"
              >
                Price/Premium
              </div>
            </div>
            <div
              class="d-md-none text-end flex-fill"
            >
              <img
                alt="Labore Inc."
                class="cardImageMob"
                src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
                title="Labore Inc."
              />
            </div>
          </section>
        </div>
        <img
          alt="Labore Inc."
          class="cardImageDesktop d-none d-lg-block ms-4"
          src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
          title="Labore Inc."
        />
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="card  rounded d-flex my-4 p-4"
      data-testid="item-click"
    >
      <div
        class="flex-fill"
      >
        <section
          class="cardTitleHolder border-bottom d-flex justify-content-between"
        >
          <i
            class="d-none d-md-block rounded-circle me-4 flex-shrink-0 flex-grow-0 cardIcon "
            data-testid="arrow"
          />
          <div
            class="flex-fill"
          >
            <h3
              class="cardTitle mb-0"
            >
              London to Paris
            </h3>
            <div
              class="cardSubTitle"
            >
              TRAVEL-COVER-INS
               | 
              Baggage Cover; Medical Cover; Cancellation Cover
            </div>
          </div>
          <img
            alt="Labore Inc."
            class="cardImageTablet d-none d-md-block d-lg-none"
            src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
            title="Labore Inc."
          />
        </section>
        <section
          class="cardDetails d-flex"
        >
          <div
            class="d-none d-md-block"
          >
            <div
              class="cardTitleDetail"
            >
              10-MAY-2019
            </div>
            <div
              class="cardSubTitleDetail"
            >
              Payment date
            </div>
          </div>
          <div>
            <div
              class="cardTitleDetail"
            >
              16-JUL-2019
               to 
              18-FEB-2019
            </div>
            <div
              class="cardSubTitleDetail"
            >
              Coverage dates
            </div>
          </div>
          <div
            class="d-none d-md-block"
          >
            <div
              class="cardTitleDetail"
            >
              AUD $106.65
            </div>
            <div
              class="cardSubTitleDetail"
            >
              Price/Premium
            </div>
          </div>
          <div
            class="d-md-none text-end flex-fill"
          >
            <img
              alt="Labore Inc."
              class="cardImageMob"
              src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
              title="Labore Inc."
            />
          </div>
        </section>
      </div>
      <img
        alt="Labore Inc."
        class="cardImageDesktop d-none d-lg-block ms-4"
        src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
        title="Labore Inc."
      />
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
